09_json/mohitnag/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/anz-bank/go-course/09_json/mohitnag/pkg/puppy"
+	"github.com/anz-bank/go-course/09_json/mohitnag/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	out  io.Writer = os.Stdout
+	app            = kingpin.New("puppyStore", "Puppy Store")
+	args           = os.Args[1:]
+)
+
+func main() {
+	mapStore := store.MapStore{}
+	syncStore := store.SyncStore{}
```

Comment:
nit: I would move the creation of the stores after the parsing of the command line args. It makes very little difference in this case, but if there was a different sort of store, such as a SQL store, 1) you may need the command line flags to initialise the store (such as a database connection string) and 2) if there is a failure to parse the command line args, there is no point creating the store.

As I said, it's very minor and I'm not asking you to make this change - I'm just giving you something to think about and explain how I would do it.

Reply:
it makes sense. I moved  the creation of the stores after parsing CLI args